packages/@react-spectrum/searchwithin/src/SearchWithin.tsx

```diff
@@ -60,20 +81,34 @@ function SearchWithin(props: SpectrumSearchWithinProps, ref: FocusableRef<HTMLEl
     isRequired,
     label: null,
     isQuiet: false,
-    'aria-labelledby': labelProps.id || ariaLabel,
-    validationState: null
+    validationState: null,
+    description: null,
+    errorMessage: null,
+    descriptionProps: null,
+    errorMessageProps: null
   };
+
   let searchFieldClassName = classNames(styles, 'spectrum-SearchWithin-searchfield');
   let pickerClassName = classNames(styles, 'spectrum-SearchWithin-picker');
+
   let slots = {
-    searchfield: {UNSAFE_className: searchFieldClassName, ...fieldProps, ...defaultSlotValues},
-    picker: {UNSAFE_className: pickerClassName, menuWidth, align: 'end', ...defaultSlotValues}
+    searchfield: {
+      ...defaultSlotValues,
+      UNSAFE_className: searchFieldClassName,
+      // Apply aria-labelledby of group or the group id to searchfield. No need to pass the group id (we want a new one) and aria-label (aria-labelledby will suffice)
+      'aria-labelledby': labelledBy
+    },
+    picker: {
+      ...defaultSlotValues,
+      id: pickerId,
+      UNSAFE_className: pickerClassName,
+      menuWidth,
+      align: 'end',
+      'aria-label': formatMessage('searchWithin'),
+      'aria-labelledby': `${labelledBy} ${pickerId}`
```

Comment:
I still wonder if this should be `${labelledBy} ${pickerId}` or it should point to the `label id + pickerId` instead if an external label exists. As it is right now it feels like it doubles up on `formatMessage('searchWithin')` and `formatMessage('search')` due to `labelledBy` containing the id of the label and the id of the group (due to the group's self reference for its `aria-label`). See the screenshot below for the `label` case
![image](https://user-images.githubusercontent.com/9661127/141031802-487c30ad-6c20-45d3-b8fa-33355f404c46.png)

and the `no label` case:
![image](https://user-images.githubusercontent.com/9661127/141032168-a4f3c7bd-5be3-455d-a2b8-3ff66910006a.png)

It makes sense to do `${labelledBy} ${pickerId}` if a custom `aria-label` is provided though like in the `no visible label case`:
![image](https://user-images.githubusercontent.com/9661127/141032493-591e5e2b-1a3f-4c7c-a742-6dec6a1a086e.png)
 
Thoughts?

cc @majornista

Reply:
The reason why the `aria-labelledby` for the picker includes the `pickerId` is to have the announcement combine the content of the `aria-label` with the other stuff `aria-labelledby` points to. If the `pickerId` isn't included in the list of `aria-labelledby`, the `aria-label` on the picker won't be announced at all since `aria-labelledby` overrides all other name sources for an element